pkg/ebpf/uprobes/procfs.go

```diff
@@ -48,14 +54,26 @@ func waitUntilSucceeds[T any](p *ProcInfo, procFile string, readFunc func(string
 	var result T
 	err := errIterStart
 	end := time.Now().Add(procFSUpdateTimeout)
+	failedTimes := 0
 
 	for err != nil && end.After(time.Now()) {
 		result, err = readFunc(filePath)
 		if err != nil {
+			failedTimes++
 			time.Sleep(procFSUpdateInterval)
 		}
 	}
 
+	if err == nil {
+		if failedTimes > 0 {
+			readTemporaryFail.Add(int64(failedTimes))
+		}
+
+		readSuccess.Add(1)
+	} else {
+		readFail.Add(1)
```

Comment:
why having 2 metrics, over a single metric with `tag:success` and `tag:failed`?

Reply:
No reason, changed.